components/server/src/analytics.ts

```diff
@@ -38,10 +45,20 @@ export async function trackSignup(user: User, request: Request, analytics: IAnal
     });
 }
 
-function fullIdentify(user: User, request: Request, analytics: IAnalyticsWriter) {
-    //makes a full identify call for authenticated users
+async function fullIdentify(
+    user: User,
+    request: Request,
+    analytics: IAnalyticsWriter,
+    subscriptionService?: SubscriptionService,
+) {
+    // makes a full identify call for authenticated users
     const coords = request.get("x-glb-client-city-lat-long")?.split(", ");
     const ip = request.get("x-forwarded-for")?.split(",")[0];
+    var subscriptionIDs: string[] | undefined = undefined;
+    const subscriptions = await subscriptionService?.getNotYetCancelledSubscriptions(user, new Date().toISOString());
+    if (subscriptions) {
+        subscriptionIDs = subscriptions.map((sub) => sub.planId).filter((x) => typeof x === "string") as string[];
+    }
```

Comment:
```suggestion
    var subscriptionIDs: string[] = [];
    const subscriptions = await subscriptionService?.getNotYetCancelledSubscriptions(user, new Date().toISOString());
    if (subscriptions) {
        subscriptionIDs = subscriptions.filter(sub => !!sub.planId).map((sub) => sub.planId!);
    }
```
We can actually simplify a bit further by making `subscriptionIDs` the empty list initially, which is a good default value when it's missing.

The actual mapping does filter first, to reduce the dataset and then does the map second to extract the IDs. It uses `!` to say it's definitely there as sugar because TS progressive type-check can't infer we've filtered out the items which are not set.

Reply:
ah yes, this is much more elegant! thanks!